milestone2.ts

```diff
@@ -125,32 +130,34 @@ let kyle: Volunteer = {
 /* Lets make a clone of Kyle above using the spread operator and assign it to kyleClone
  */
 type CopyVolunteer = (vol: Volunteer) => Volunteer;
-let copyVolunteer: CopyVolunteer; // Code here
+let copyVolunteer: CopyVolunteer = vol => {return {...vol}}; // Code here
 
-// let kyleClone: Volunteer = copyVolunteer(kyle);
-//console.log(kyleClone);
+let kyleClone: Volunteer = copyVolunteer(kyle);
+console.log(kyleClone);
 
 // Question 6: Spread Operator Part 2
 /* Next, lets use the spread operator to update your kyleClone object with the updatedLocation defined below and assign it to kyleNew
  */
 let updatedLocation = { city: "Seattle", state: "WA" };
-let kyleNew: Volunteer;
+//let kyleNew: Volunteer;
 type UpdateVolunteer = (
   vol: Volunteer,
   updates: Partial<Volunteer>
 ) => Volunteer;
-let updateVolunteer: UpdateVolunteer; // Code here
+let updateVolunteer: UpdateVolunteer = (vol, updates) => {
+  return {...vol, ...updates}
+}; // Code here
 
-// let kyleNew = updateVolunteer(kyleClone);
-//console.log(kyleNew);
+let kyleNew: Volunteer = updateVolunteer(kyleClone, updatedLocation);
+console.log(kyleNew);
 
 // Question 7: Object Destructuring
 /* Now that we have our updated kyle, lets use object destructuring to get his name, age, and city
  */
 
 type GetVolunteerInfo = (vol: Volunteer) => String;
 let getVolunteerInfo: GetVolunteerInfo = (vol) => {
-  let name, age, city; // Code here
+  let [name, age, city] = [vol.name, vol.age, vol.city]; // Code here
```

Comment:
Looks like you forgot to use object destructuring here, can you resubmit with object destructuring on this question? An example of object destructuring can be found in the demo from [Bootcamp Session 3](https://h4i.notion.site/Intro-to-Javascript-83f29480d7c14b0c9163212b96f6ab0f)